git_machete/cmd.py

```diff
@@ -1515,6 +1516,84 @@ def flush_caches(self) -> None:
+        pr: GitHubPullRequest = check_pr_already_created_by_number(pr_no, all_prs)
+
+        if not pr:
+            raise MacheteException(f"PR number {pr_no} is not found in repository {repo}")
+        debug('checkout_github_pr', f'found {pr}')
+
+        current_branch: Optional[str] = self.__git.get_currently_checked_out_branch_or_none()
+        if current_branch == 'PR_' + str(pr.number):
+            self.__git.checkout(pr.head)
+
+        self.__git.fetch_pull_request(remote, pr.number)
+        debug('checkout_github_pr', f'fetched PR into branch PR_{pr}')
+
+        self.__git.fetch_remote(remote)
+
+        path: List[str] = self.get_path(remote, pr.head)
+        upstream_prs: List[GitHubPullRequest] = []
+        if len(path) > 1:
+            for index, branch in enumerate(path):
+                try:
+                    possible_pr: GitHubPullRequest = GitHubPullRequest(0, '', path[index + 1], branch, '')
+                    pr_found = check_pr_already_created(possible_pr, all_prs)
+                    if pr_found and pr_found != pr and pr_found.base != branch:
+                        upstream_prs.append(pr_found)
+                except IndexError:
+                    break
+
+        if upstream_prs:
+            reversed_path = path[::-1]  # need to add from root do down
+            for index, branch in enumerate(reversed_path):
+                if branch not in self.managed_branches:
+                    self.__cli_opts.opt_onto = reversed_path[index - 1]
+                    self.add(branch)
+
+            self.sync_annotations_to_github_prs()
+
+        self.__git.checkout(f'PR_{pr.number}')
+        print(fmt(f"Pull request #{pr.number} checked out at local branch PR_{pr.number}"))
+
+    def get_path(self, remote: str, last_branch: str) -> List[str]:
+        def get_parent_remote_branch(branch: str) -> str:
+            checked: Dict[int, str] = {}
+            base: List[str] = self.__git.get_stripped_remote_branches(remote)
+            for item in path:
+                base.remove(item)
+
+            for b in base:
+                common_commit_hash: str = self.__git.popen_git("merge-base", '/'.join([remote, b]), '/'.join([remote, branch])).rstrip()
+                commit_timestamp: int = int(self.__git.get_commit_timestamp(common_commit_hash))
+                checked[commit_timestamp] = b
+
+            parent: str = checked[max(checked.keys())]
+            branch_first_commit_time: int = int(self.__git.get_first_reflog_entry_timestamp(remote, branch))
+
+            while int(self.__git.get_first_reflog_entry_timestamp(remote, parent)) > branch_first_commit_time:
+                del checked[max(checked.keys())]
+                parent = checked[max(checked.keys())]
+            return parent  # the youngest common commit found is considered to be from a parent branch
+            # but there is case unhadled when 2 different branches has the youngest common commit (which branch to take?)
```

Comment:
When parent branch has another child branch and child branch doesn't have any commits